frontend/components/Profile/Profile.jsx

```diff
@@ -0,0 +1,280 @@
+import React from 'react';
+import { Divider, Button } from '@blueprintjs/core';
+import PropTypes from 'prop-types';
+import gql from 'graphql-tag';
+import { Mutation } from 'react-apollo';
+import InputField from '../commons/InputField';
+import FormContainer from '../FormContainer';
+import useFormInput from '../hooks/useFormInput';
+import samePassword from '../../lib/samePassword';
+
+const UPDATE_ME = gql`
+  mutation UPDATE_PROFILE($user: UserUpdateInput!) {
+    updateProfile(user: $user) {
+      id
+      name
+    }
+  }
+`;
+
+const Profile = ({ user }) => {
+  const { address, bankDetails } = user;
+  const name = useFormInput(user.name);
+  const email = useFormInput(user.email);
+  const street = useFormInput(address ? address.street : '');
+  const city = useFormInput(address ? address.city : '');
+  const country = useFormInput(address ? address.country : '');
+  const zipCode = useFormInput(address ? address.zipCode : '');
+  const iban = useFormInput(bankDetails ? bankDetails.iban : '');
+  const bic = useFormInput(bankDetails ? bankDetails.bic : '');
+
+  const password = useFormInput('');
+  const passwordRepeat = useFormInput('');
+
+  const onSubmit = (e, save) => {
+    e.preventDefault();
+    save();
+  };
+
+  return (
+    <FormContainer>
+      <h2>General</h2>
+      <Divider />
+      <br />
+      <Mutation
```

Comment:
Did you try using a HOC for the GraphQL queries to make this component more readable? It can be passed in as props. I can try it and commit if you like?